x/auth/ante/sigverify.go

```diff
@@ -290,7 +236,9 @@ func (svd SigVerificationDecorator) authenticate(ctx sdk.Context, tx sdk.Tx, sim
 		return nil
 	}
 
-	return svd.increaseSequence(ctx, acc)
+	// update sequence; it will also set the account's key
+	// for the first time if it was not populated before.
+	return svd.increaseSequenceAndUpdateAccount(ctx, acc)
```

Comment:
Why the name change? Also, `increaseSequenceAndUpdateAccount` doesn't update the pubkey from what I can tell.

Reply:
it's an unfortunate consequence of the pubkey being part of the account struct... 

in fact the pubkey is updated here: https://github.com/cosmos/cosmos-sdk/pull/19093/files#diff-b94baf116534b68c0bc48cb6faa0b88e1bd2b12d7a2c2dd42b579cd64a256db8R379

but persisted to state by the increaseSequenceAndUpdateAccount method.

Reply:
This is confusing and possibly a footgun. I would consider keeping `increaseSequence` and then having an explicit `SetAccount` call here.

Reply:
kk, i'll split the update part from increase sequence

Reply:
I refactored both the increaseSequence and setPubKey method not to accept the sdk.Context, so it's clear that no state updates are being applied until the very end. 

lmk if this is good